<a href="https://colab.research.google.com/github/gandhias-a11y/colab/blob/main/hw1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **PROBLEM 1**

Fair means that the expected value is equal to $0$. The probability of winning craps, rolling a seven, is $\frac{6}{36} = \frac{1}{6}$. The probability of not winning, rolling anything other than seven, is $\frac{30}{36} = \frac{5}{6}$. Thus, the expected value is $(4 \cdot \frac{1}{6} - 1 \cdot \frac{5}{6}) = \frac{4}{5} = \frac{5}{6} = -\frac{1}{6}$. Since the expected value is not zero, this is not fair. The game would be fair if the house paid $5$ chips instead of $4$ if you rolled a seven.

In [ ]:
import random
wallet = 0
runtime = 1000000

for i in range(runtime):
  dice_1 = random.randint(1,6)
  dice_2 = random.randint(1,6)
  crap_sum = dice_1 + dice_2
  if crap_sum == 7:
    wallet += 4
  else:
    wallet -= 1

print(wallet/runtime)

-0.16708


The above code essentially models what your pool of money would look like after many, many rounds of craps. At the end of each round, the amount of chips you have increases by four or decreases by one. After these many rounds, you are left with a negative amount of chips per round which means that you did not come out with a positive amount of chips compared to when you started. This value of chips per round is another way of calculating the expected value as in the first part of Problem $1$.

# **PROBLEM 2**

In [ ]:
count = 0

for dice1 in range(1,7):
  for dice2 in range(1, 7):
    for dice3 in range(1,7):
      sum = dice1 + dice2 + dice3
      if sum not in [3, 5, 7, 11, 13,17]:
        count +=1

print(count/216)

0.6620370370370371


In [ ]:
import random

count = 0
times = 1000000

for i in range(times):
  d1 = random.randint(1, 6)
  d2 = random.randint(1,6)
  d3 = random.randint(1,6)
  sum = d1 +d2 +d3
  if sum not in [3, 5, 7, 11, 13, 17]:
    count += 1

print(count/times)



0.662592


In the above section of code, I simulated what the result, sum, of rolling three dice would be one million times. The goal was to see the ratio of how many times the sum of the three dice would be prime to the total number of simulations (one million). The result was $0.663$ which shows that the result of the sum being a prime number came roughly two thirds of the time.

# **PROBLEM 3**

In [ ]:
lockers = [''] * 100

for student in range(100):
  for locker in range(len(lockers)):
    if (locker + 1) % (student + 1) == 0:
      if lockers[locker] == '':
        lockers[locker] = '*'
      else:
        lockers[locker] = ''
  print(' '.join(lockers))


* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
*  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  * 
*    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *    * * *   
*   * * * * *   *  *   * * * * *   *  *   * * * * *   *  *   * * * * *   *  *   * * * * *   *  *   * * * * *   *  *   * * * * *   *  *   * * * * *   *  *   *
*   *  * * *  * *  *  * * * * *    *     * *  * *     *    * * * * *  *  * *  * * *  *   * * *   *  * * *  * *  *  * * * * *    *     * *  * *     *   
*   *   * *  * * * *  * * *  *    * *    * * * * *    * *    *  * * *  * * * *  * *   *   *  *   *   * *  * * * *  * * *  *    * *    * * * * *    * *   
*   *    *  * * * * * * * *  *  *  * 

The above program evalutates each stage of the described scenario: there are a hundred lockers for a hundred students. The first student opens every locker, the second student closes every other locker, the third student goes to every third locker and opens it or closes it depending on its current status, and so on. The final line printed by the program displays which lockers are open (conveyed by the asterisk) after the hundreth student passes through. The program shows that the 1st, 4th, 9th, 16th, 25th, 36th, 49th, 64th, 81st, and 100th lockers are open at the end.